runtime/drivers/duckdb/connectors.go

```diff
@@ -189,3 +182,36 @@ func fileSize(paths []string) int64 {
 	}
 	return size
 }
+
+func resolveLocalPath(env *connectors.Env, path, sourceName string) (string, error) {
+	path, err := fileutil.ExpandHome(path)
+	if err != nil {
+		return "", err
+	}
+
+	repoRoot := env.RepoRoot
+	if repoRoot != "" {
+		repoRoot, err = filepath.Abs(repoRoot)
+		if err != nil {
+			return "", err
+		}
+	}
```

Comment:
1. The repo root could also contain a `~`, so might need a `fileutil.ExpandHome`
2. Why can the `repoRoot` be `""`?
3. I think it would be neater if the `repo.Root()` func guaranteed returning a non-empty, absolute path, so these downstream checks are not necessary

Reply:
1. That is already done before this function is being called. 
2. To be very frank, even I am not aware of the case where `repoRoot` can be `""`. The existing code handles that case so I imagined its some case that I am not aware of.
3. Yes. That can be done as well.

Reply:
On reflection, maybe the `repoRoot == ""` case can occur when the repo is the current `pwd`, so any relative path will just work. I guess doing `filepath.Abs("")` should return the `pwd`, so if we make `repo.Root()` guarantee an absolute path, then we would not need to think about this case anymore